src/RA_Dlg_GameLibrary.cpp

```diff
@@ -288,13 +291,11 @@ void Dlg_GameLibrary::ThreadedScanProc()
 
     while (FilesToScan.size() > 0)
     {
-        mtx.lock();
-        if (!Dlg_GameLibrary::ThreadProcessingAllowed)
         {
-            mtx.unlock();
-            break;
+            std::scoped_lock lock{ ra::mtx };
```

Comment:
Since I had to prefix `mtx `anyway, it looked like a good opportunity to put in RAII management for the mutex.